aws/s2n-tls
# fix: open file decriptors

### Resolved issues:

Partially solve #4005.

### Description of changes: 

* Use `s2n_io_pair_close` and `s2n_io_pair_close_one_end` to close all opened `AF_UNIX` socket.
* Clean up all of opened `/dev/urandom`.

### Call-outs:

* I didn't add a test in this PR. The test to detect all opened fds will be in a separate PR.

### Testing:

* Test locally. 
    * Add `--track-fds=yes` to CTest memcheck and direct all Valgrind output to `MemoryTester.log` files.
    * Search in VS Code for Open File Descriptors of `AF_UNIX` or `/dev/urandom`. Nothing was captured.
    * Manually go through every log files to see if there are anything that are missed.
    * This is similar to the check in `./codebuild/bin/test_exec_leak.sh`.
  https://github.com/aws/s2n-tls/blob/6bb195c23fd1ecf623dece3b43644e7c9a9e0e20/codebuild/bin/test_exec_leak.sh#L82-L97

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
